docker/install-dynamic-plugins.py

```diff
@@ -789,6 +792,87 @@ def wait_for_lock_release(lock_file_path):
+            local_dir = os.path.join(tmp_dir, 'catalog-index-oci')
+
+            # Download the OCI image using skopeo
+            result = subprocess.run(
+                [skopeo_path, 'copy', image_url, f'dir:{local_dir}'],
+                capture_output=True,
+                text=True
+            )
+            if result.returncode != 0:
+                print(f"WARNING: Failed to download catalog index image: {result.stderr}", flush=True)
+                return None
+
+            manifest_path = os.path.join(local_dir, 'manifest.json')
+            if not os.path.isfile(manifest_path):
+                print("WARNING: manifest.json not found in catalog index image", flush=True)
+                return None
+
+            with open(manifest_path, 'r') as f:
+                manifest = json.load(f)
+
+            print(f"\t==> Extracting catalog index layers", flush=True)
+            max_entry_size = int(os.environ.get('MAX_ENTRY_SIZE', 20000000))
+
+            for layer in manifest.get('layers', []):
+                layer_digest = layer.get('digest', '')
+                if not layer_digest:
+                    continue
+
+                (_sha, filename) = layer_digest.split(':')
+                layer_file = os.path.join(local_dir, filename)
+                if not os.path.isfile(layer_file):
+                    print(f"\t==> WARNING: Layer file {filename} not found", flush=True)
+                    continue
+
+                print(f"\t==> Extracting layer {filename}", flush=True)
+                with tarfile.open(layer_file, 'r:*') as tar:
+                    for member in tar.getmembers():
+                        # Security checks
+                        if member.size > max_entry_size:
+                            print(f"\t==> WARNING: Skipping large file {member.name} in catalog index", flush=True)
+                            continue
+                        if member.islnk() or member.issym():
+                            realpath = os.path.realpath(os.path.join(catalog_index_temp_dir, *os.path.split(member.linkname)))
+                            if not realpath.startswith(catalog_index_temp_dir):
+                                print(f"\t==> WARNING: Skipping link outside archive: {member.name}", flush=True)
+                                continue
+                        tar.extract(member, path=catalog_index_temp_dir, filter='tar')
+
+        default_plugins_file = os.path.join(catalog_index_temp_dir, 'dynamic-plugins.default.yaml')
+        if os.path.isfile(default_plugins_file):
+            print(f"\t==> Successfully extracted catalog index with dynamic-plugins.default.yaml", flush=True)
+            return default_plugins_file
+        else:
+            print(f"\t==> Catalog index extracted but dynamic-plugins.default.yaml not found", flush=True)
+            return None
+
+    except Exception as e:
+        print(f"WARNING: Error extracting catalog index: {e}", flush=True)
+        return None
```

Comment:
Exceptions should be propagated and not be ignored IMO. It can mean that there are some issues with the catalog index image, so I'd suggest surfacing those earlier by aborting the init container.